Feature/#4 navbar

bomii1 opened the pull request:
## #️⃣ 연관된 이슈

Closes: #4 

## #️⃣ 작업 내용
- Layout.tsx 전체 배경색 변경: bg-white/70 -> bg-white/40
- 사이드바 구현

<img width="3024" height="1890" alt="image" src="https://github.com/user-attachments/assets/2f2e6806-8ede-4784-8702-7d8cdcedfb67" />
<img width="3024" height="1890" alt="image" src="https://github.com/user-attachments/assets/f377cf89-b853-4053-aab6-852dbd22491f" />

bomii1 committed:
feat: 사이드바, 메인 레이아웃

bomii1 committed:
feat: 사이드바

meteorqz6 commented:
관련 이슈 내용 보충 부탁드립니당
선택 사항 말고 필수 사항은 채워주세요!

meteorqz6 commented on package.json:
```diff
@@ -21,10 +21,13 @@
     "react-dom": "^19.1.1",
+    "react-icons": "^5.5.0",
     "react-router": "^7.9.4",
+    "react-router-dom": "^7.9.4",
```
공식 문서를 보니까 React Router v7에서는 패키지가 단순화되었기 때문에 더 이상 "react-router-dom"이 필요하지 않고, 이제 모든 것을 "react-router"에서 가져올 수 있다고 합니다.

아래와 같이 수정하면 어떨까요?
1. package.json에서 react-router-dom 의존성 삭제
2. 모든 import ... from "react-router-dom" 구문을 import ... from "react-router"로 변경

이렇게 하면 프로젝트 의존성을 깔끔하게 유지하고 향후 발생할 수 있는 혼란을 미리 방지할 수 있을 것 같습니다!

공식문서 참조 링크: https://reactrouter.com/upgrading/v6#upgrade-to-v7

meteorqz6 commented on src/components/layout/Sidebar.tsx:
```diff
@@ -0,0 +1,103 @@
+import { BiHomeAlt } from "react-icons/bi";
+import { LuCalendar, LuSearch } from "react-icons/lu";
+import { IoFileTraySharp, IoLogOutOutline } from "react-icons/io5";
+import { NavLink } from "react-router-dom";
```
"react-router"에서 import하면 될 것 같아요

meteorqz6 replied:
```suggestion
import { NavLink } from "react-router";
```

meteorqz6 commented on src/components/layout/Sidebar.tsx:
```diff
@@ -0,0 +1,103 @@
+      {/* 프로필 영역 */}
+      <div
+        className={clsx(
+          "flex-[1] flex items-center border-b border-[#EFEFEF] px-2 py-2",
```
```suggestion
          "flex-[1] flex items-center border-b border-[#EFEFEF] px-3 py-2",
```

meteorqz6 commented on src/components/layout/Sidebar.tsx:
```diff
@@ -0,0 +1,103 @@
+      <div
+        className={clsx(
+          "flex-[1] flex items-center border-b border-[#EFEFEF] px-2 py-2",
+          isOpen ? "justify-between mr-2" : "justify-center"
```
```suggestion
          isOpen ? "justify-between" : "justify-center"
```

meteorqz6 replied:
border-b border-[#EFEFEF]가 mr-2 때문에 애매하게 끊어져 보여서 없애는 좋을 것 같아요

bomii1 commented:
> 관련 이슈 내용 보충 부탁드립니당 선택 사항 말고 필수 사항은 채워주세요!

완료했습니다. 확인하고 리뷰 부탁드려요.

bomii1 committed:
Update src/components/layout/Sidebar.tsx

Co-authored-by: Nam Yooseong <102887277+meteorqz6@users.noreply.github.com>

bomii1 committed:
Update src/components/layout/Sidebar.tsx

Co-authored-by: Nam Yooseong <102887277+meteorqz6@users.noreply.github.com>

bomii1 committed:
Update src/components/layout/Sidebar.tsx

Co-authored-by: Nam Yooseong <102887277+meteorqz6@users.noreply.github.com>

bomii1 committed:
react-router로 통일

bomii1 committed:
pakage.json 수정

- dependencies에서 react-router-dom 삭제
- devDependencieds 에서 잘못 삭제한 @types/react-dom" 추가

bomii1 committed:
package.json 수정
